topiary-cli/src/configuration.rs

```diff
@@ -1,148 +1,379 @@
+use clap::ValueEnum;
 use directories::ProjectDirs;
-use std::{env::current_dir, path::PathBuf};
+use indoc::formatdoc;
+use itertools::Itertools;
+use std::{env::current_dir, fmt, path::PathBuf};
 use topiary::{default_configuration_toml, Configuration};
 
 use crate::error::{CLIResult, TopiaryError};
 
-pub fn parse_configuration(
-    config_override: Option<PathBuf>,
-    config_file: Option<PathBuf>,
-) -> CLIResult<Configuration> {
-    user_configuration_toml(config_override, config_file)?
-        .try_into()
-        .map_err(TopiaryError::from)
+type Annotations = String;
+
+/// Collation mode for configuration values
+// NOTE The enum variants are in "natural" order, rather than
+// sorted lexicographically, for the sake of the help text
+#[derive(Clone, Debug, ValueEnum)]
+pub enum CollationMode {
```

Comment:
Would `CombinationMode` be a valid alternative? I didn't know what "Collation" meant, although that could just be me ;)

Reply:
"Collation" is a term used in databases. It has a _slightly_ different meaning in that field: it's more to do with sort order and case/accent sensitivity, but has a very similar effect and I figured it applied here quite naturally. (It also corresponds, IMO, with the natural definition of the word.)

"Combination" feels not-quite-right because overriding is not really combining. (But I appreciate that this is a pretty weak argument!)